tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/RealMetricName.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics.registry;
+
+import com.google.common.collect.ImmutableSortedMap;
+import java.util.SortedMap;
+
+final class RealMetricName implements MetricName {
+    private static final SortedMap<String, String> EMPTY = prehash(ImmutableSortedMap.of());
+    private final String safeName;
+    private final SortedMap<String, String> safeTags;
+    private final int hashCode;
+
+    private RealMetricName(String safeName, SortedMap<String, String> safeTags) {
+        this.safeName = safeName;
+        this.safeTags = safeTags;
+        this.hashCode = computeHashCode();
+    }
+
+    private int computeHashCode() {
+        int hash = 5381;
+        hash += (hash << 5) + safeName().hashCode();
+        hash += (hash << 5) + safeTags().hashCode();
```

Comment:
just curious why djb2 hash algorithm?

Reply:
it's the one in the immutables generated code

Reply:
copy-pasted :P